File: AUGUST-2025/02-08-2025/p2561.scala
object Solution {
  def minCost(basket1: Array[Int], basket2: Array[Int]): Long = {
    import scala.collection.mutable
    val count = mutable.TreeMap[Long, Long]() 
    basket1.foreach(a => count.update(a.toLong, count.getOrElse(a.toLong, 0L) + 1L))
    basket2.foreach(a => count.update(a.toLong, count.getOrElse(a.toLong, 0L) - 1L))
    val small = count.firstKey
    if (count.values.exists(_ % 2 != 0)) return -1
    val swaps = count.flatMap { case (a, v) =>
      Seq.fill((math.abs(v) / 2).toInt)(a)
    }.toList
    swaps.view.take(swaps.size / 2).map(swap => math.min(swap, small * 2)).sum
  }
}

File: AUGUST-2025/02-08-2025/p2561.swift
class Solution {
    func minCost(_ b1: [Int], _ b2: [Int]) -> Int {
        var m: [Int: Int] = [:]
        for c in b1 {
            m[c, default: 0] += 1
        }
        
        for c in b2 {
            m[c, default: 0] -= 1
        }
        
        var swaps = 0, res = 0
        
        for (c, cnt) in m {
            if cnt % 2 == 1 {
                return -1
            }
            swaps += max(0, cnt / 2)
        }
        
        let m1 = m.sorted(by: { $0.key < $1.key })
        
        for (c, cnt) in m1 {
            let take = min(swaps, abs(cnt) / 2)
            res += take * min(c, m1.first!.key * 2)
            swaps -= take
        }
        
        return res
    }
}